File: src/main/java/com/atlaspharmacy/atlaspharmacy/users/domain/MedicalStaff.java
package com.atlaspharmacy.atlaspharmacy.users.domain;

import com.atlaspharmacy.atlaspharmacy.generalities.DTO.AddressDTO;
import com.atlaspharmacy.atlaspharmacy.generalities.domain.Address;
import com.atlaspharmacy.atlaspharmacy.generalities.mapper.AddressMapper;
import com.atlaspharmacy.atlaspharmacy.users.DTO.AuthorityDTO;
import com.atlaspharmacy.atlaspharmacy.users.DTO.MedicalStaffDTO;
import com.atlaspharmacy.atlaspharmacy.users.domain.enums.Gender;
import com.atlaspharmacy.atlaspharmacy.users.domain.enums.Role;
import com.atlaspharmacy.atlaspharmacy.users.mapper.AuthorityMapper;
import org.hibernate.annotations.Proxy;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "medicalstaff")
@DiscriminatorValue(value = Role.Values.MedicalStaff)
@Proxy(lazy = false)
public class MedicalStaff extends User {
    private String licenseNumber;

    public  MedicalStaff() {}
    public MedicalStaff(String licenseNumber) {
        this.licenseNumber = licenseNumber;
    }

    public MedicalStaff(String name, String surname, Date dateOfBirth,
                           String phoneNumber, String email, String password,
                           Gender gender, Address address, String role,
                           List<Authority> authorities, boolean firstTimeChanged,String licenseNumber){
            super(name,surname,address, dateOfBirth, phoneNumber, gender, role, authorities);
            this.licenseNumber=licenseNumber;
    }

    @Override
    public String getUsername() {
        return getEmail();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }

    public String getLicenseNumber() {
        return licenseNumber;
    }

    public void setLicenseNumber(String licenseNumber) {
        this.licenseNumber = licenseNumber;
    }
}

File: frontend/src/app/model/users/medicalStaff.ts
import {User} from "@app/model/users/user";
import {Gender} from "@app/model/users/patient/gender";
import {Address} from "@app/model/address/address";
import {Role} from "@app/model/users/role";
import {Pharmacy} from "@app/model/pharmacy/pharmacy";

export class MedicalStaff {
  public id:Number;
  public name : String;
  public surname : String;
  public dateOfBirth : Date;
  public phoneNumber : String;
  public email : String;
  public password : String;
  public gender : Gender;
  public address : Address;
  public role : Role;
  public authorities : Number[];
  public firstTimeChanged : boolean;
  public licenceNumber : String

  constructor(id:Number, name:String, surname : String, dateOfBirth: Date
              , phoneNumber : String, email : String, password : String
              , gender: Gender, address : Address, role : Role, autorities : Number[]
              , firstTimeChanged:boolean,licenceNumber:String){
    this.id=id;
    this.name = name;
    this.surname = surname;
    this.dateOfBirth = dateOfBirth;
    this.phoneNumber = phoneNumber;
    this.email = email;
    this.password = password;
    this.gender = gender;
    this.address = address;
    this.role = role;
    this.authorities = autorities;
    this.firstTimeChanged = firstTimeChanged;
    this.licenceNumber = licenceNumber;
  }

}
